flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/rules/logical/FlinkJoinToMultiJoinRuleTest.scala

```diff
@@ -268,4 +291,19 @@ class FlinkJoinToMultiJoinRuleTest extends TableTestBase {
         """.stripMargin
     util.verifyRelPlan(sqlQuery)
   }
+
+  @Test
+  def testMultiLeftOuterJoinWithAllKeyInLeft: Unit = {
```

Comment:
is any test with semi/anti to verify the change ?

Reply:
semi and anti now can not join reorder, so it will not go into `onMatch()` method because of `origJoin.getJoinType().projectsRight()` in `matches()`. So I think the branch `inputNullGenFieldList.isEmpty()` in `canCombine()` is useless and I will remove it.